robo/rosafe/matlab_client.txt

```diff
@@ -3,4 +3,211 @@
+
+  % Manager
+  mObj = manager_RosAFE(dObj);
+
+Then, you can make a request with exactly the same code as the |AFE|:
+
+.. code-block:: Matlab
+
+  % Request with default parameters
+  mObj.addProcessor('ild');
+  mObj.addProcessor('ratemap');
+
+Again, creating a parameter structure remains exactly the same as with the |AFE|:
+
+.. code-block:: Matlab
+
+  % Parameter settings
+  pp_bRemoveDC   = 1;
+  pp_cutoffHzDC   = 5000;
+
+  % Parameter structure
+  par = genParStruct('pp_bRemoveDC',pp_bRemoveDC, 'pp_cutoffHzDC',pp_cutoffHzDC);
+
+  % Request with seted parameters
+  mObj.addProcessor('time',par); 
+
+On-the-fly parameter modifications can be requested with the ``modifyParameter`` function:
+
+.. code-block:: Matlab
+  
+  % turning off the DC removal filter
+  nameOfProcessor = 'time_1'; % this name can be found by the getParameters function.
+  nameOfParameter = 'pp_cutoffHzDC';
+  newValue = '6000';
+  mObj.modifyParameter( nameOfProcessor, nameOfParameter, newValue );
+
+Differently from the Matlab AFE, a processor immediately computes its output(s) when instantiated
+from the Matlab ``/rosAFE`` client. This means that as soon as data are available on its input
+(s), these are processed and the resulting audio representation is published on its corresponding
+|GenoM3| port(s), even if the proposed manager doesn't explicitly ask to actually process audio
+chunks. This was the role of the ``processChunk()`` method in the genuine AFE, while this
+method is now used to load the processed audio representation from the |Genom3| environment to the
+Matlab environment. From the user viewpoint, this slight change in the method is transparent, and
+``processChunk()`` still presents the requested audio representation as output, which will be stored 
+in the data object.
+
+.. code-block:: Matlab
+
+  % Loading all the outputs of the all processors/
+  mObj.processChunk( );
+
+A processor can be deleted by the ``deleteProcessor`` function.
+
+.. code-block:: Matlab
+
+  typeOfProcessor = 'ild';
+  orderOfProcessor = 1; % this number can be found by the getParameters function. 1 means the first ild processor.
+  mObj.deleteProcessor( typeOfProcessor, numberOfProcessor );
+
+Individual ports (i.e. individual processor output) can also be loaded to Matlab to get the data:
+
+.. code-block:: Matlab
+
+  nameOfProcessor = 'ratemap_0'; % this number can be found by the getParameters function
+  output = mObj.RosAFE.ratemapPort(nameOfProcessor);
+
+Demo
+~~~~~~~~
+
+The following script will continuously plot the results of the ILD processor coming from the RosAFE.
+
+.. code-block:: Matlab
+
+  clear all; close all; clc;
+
+  openRobotsMatlabPath = '~/openrobots/lib/matlab';
+  twoEarsPath = '~/TwoEars/AuditoryModel/TwoEars-1.2/';
+  rosAFE_matlab_Path = '~/genom_ws/rosAFE/matlabFiles';
```

Comment:
Maybe we should add a comment before these paths that the user has to adjust them?